src/include/sof/lib/dai-zephyr.h

```diff
@@ -168,6 +168,10 @@ struct dai_data {
 
 	/* llp slot info in memory windows */
 	struct llp_slot_info slot_info;
+	/* save current sampling for current dai device */
+	uint32_t sampling;
+	/* non fast mode, use memory to save mcps */
+	bool non_fast_mode;
```

Comment:
I prefer to use if (!dd->fast_mode).    Rarely use non_xxxx_xxxx.

Reply:
it will add extra cost from performance perspective, since ! need at least one more instruction operation.

Reply:
@btian1 compiler should produce equal-speed code for both variants

Reply:
that's not possible, it definitely need a ! operation to get the desired results.

Reply:
> @btian1 compiler should produce equal-speed code for both variants

@kv2019i yeah, right, in principle. Only if one of the branches is taken much more often, but Zephyr has `likely()` and `unlikely()` for that